x/ref/cmd/principal/main.go

```diff
@@ -37,8 +37,10 @@ import (
 
 var (
 	// Flags for the "blessself" command
-	flagBlessSelfCaveats caveatsFlag
-	flagBlessSelfFor     time.Duration
+	flagBlessSelf = struct {
+		Caveats caveatsFlag   `cmdline:caveat,,"package/path".CaveatName:VDLExpressionParam to attach to this blessing"`
+		For     time.Duration `cmdline:"for,0,Duration of blessing validity (zero implies no expiration)"`
```

Comment:
Double quotes in both flags need corrections. For Caveats, quotes are not matched. For For, the flag name, default value, etc. should be quoted separately.

Looks like this should be throwing a runtime error based on the test cases in flagvar_test.go.

Reply:
these are actually escaped - it looks like github is missing up the quotes!

`cmdline:"caveat,,\\"package/path\\".CaveatName:VDLExpressionParam to attach to this blessing"`

Reply:
Sorry, it should be single \ above.

Reply:
Something's wrong. Even the raw file view does not show the escapes -- https://raw.githubusercontent.com/vanadium/core/ea0900f6d751cb8809580e49d071452f5b0a1908/x/ref/cmd/principal/main.go

If you are confident it's there, go ahead and commit.

Reply:
Actually, something is definitely wrong.

https://travis-ci.org/vanadium/core/builds/461584799#L803

Reply:
Oh, sure, it's broken for other reasons, not least because it needs the other PR:
https://github.com/vanadium/go.lib/pull/27 
to be reviewed and submitted. Once that's in, I stand a chance of getting this work, but for now, it's broken because of some mixup with the previously global flags that I haven't quite tracked down yet! Thanks for taking a look!

Reply:
I had to create a new PR for flagvar to fix a bug that this revision exposes. https://github.com/vanadium/go.lib/pull/28. I'd be grateful if you could approve that one too. I don't plan on converting everything to use flagvar, just enough to make sure it works as intended. It seems a reasonable improvement to me. I expect it's a bigger win for smaller commands than this one. Thanks!